shell.nix

```diff
@@ -3,11 +3,11 @@ let
     packageOverrides = pkgs: rec {
       haskellPackages = pkgs.haskell.packages.ghc822.override {
         overrides = haskellPackagesNew: haskellPackagesOld: rec {
-          universum =
-            pkgs.haskell.lib.dontCheck
-              (haskellPackagesNew.callHackage "universum" "1.1.0" {});
-          log-classifier =
-            (haskellPackagesNew.callCabal2nix "log-classifier" ./. {});
+          universum           = pkgs.haskell.lib.dontCheck (haskellPackagesNew.callHackage "universum" "1.1.0" {});
+          concurrent-output   = pkgs.haskell.lib.dontCheck (haskellPackagesNew.callPackage ./concurrent-output.nix {});
+          # because of http://hackage.haskell.org/package/concurrent-output
+          stm                 = pkgs.haskell.lib.dontCheck (haskellPackagesNew.callPackage ./stm.nix {});
```

Comment:
Would rather just have CallHackage on everything to reduce unnecessary nix files. Tried to fix it and got an infinite recursion dependency issue. Will fix later.

Reply:
That's what I got as well so I decided to use this instead. AFAICU, infinite recursion dependency issue occurs when there is a namespace collision?